Source/Scene/ShadowMapShader.js

```diff
@@ -0,0 +1,368 @@
+/*global define*/
+define([
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Renderer/ShaderSource'
+    ], function(
+        defaultValue,
+        defined,
+        ShaderSource) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function ShadowMapShader() {
+    }
+
+    function findVarying(shaderSource, names) {
```

Comment:
Should this and helpers below be static functions on `ShaderSource` like https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Renderer/ShaderSource.js#L287-L290?